scripts/turd/turd.lua

```diff
@@ -46,7 +46,12 @@ local function on_search(search_key, gui, player)
         if sub_tech_flow then
             local tech_upgrade = tech_upgrades[element.tags.name]
             local name = tech_upgrade.master_tech.name:lower()
-            element.visible = not not (check_viewable(element, player, researched_technologies) and name:find(search_key, 1, true))
+            player.request_translation(player.force.technologies[name].localised_name)
+            translated_name = ""
+            if(storage.technology_locale[player.index] and storage.technology_locale[player.index][player.force.technologies[name].localised_name[1]]) then
+                translated_name = storage.technology_locale[player.index][player.force.technologies[name].localised_name[1]]:lower()
```

Comment:
Using `player.force.technologies[name].localised_name[1]]` is a bit fraught, I think. [1] for a locale can be, for example, `"?"` or `""` to denote a fallback or plaintext concatenation respectively ([api docs](https://lua-api.factorio.com/latest/concepts/LocalisedString.html)).

I think it might be better to index the storage by locale (`player.locale`) and prototype name, if this doesn't introduce too much complexity. e.g. `["en"]["auog-upgrade"]`